pallets/nodle-staking/src/lib.rs

```diff
@@ -388,6 +460,7 @@ pub mod pallet {
             origin: OriginFor<T>,
             validator: T::AccountId,
             amount: BalanceOf<T>,
+            unfreeze_bond: bool,
```

Comment:
I am not clear what this parameter is?

Reply:
As discussed

* Regarding argument unfreeze_bond in API

```
pub fn nominator_nominate(
    origin: OriginFor<T>,
    validator: T::AccountId,
    amount: BalanceOf<T>,
    unfreeze_bond: bool,
) -> DispatchResultWithPostInfo;

pub fn nominator_bond_more(
    origin: OriginFor<T>,
    validator: T::AccountId,
    more: BalanceOf<T>,
    unfreeze_bond: bool
) -> DispatchResultWithPostInfo;
```
As part of reconciliation, if nomination bond value is below threshold

![](https://i.imgur.com/lM14mwY.png)

nomination is removed from validator pool
and dust or residual value is moved to new variable frozen_bond ...

![](https://i.imgur.com/i0zY8Mi.png)

This value can be unlocked back as part of
nominator_nominate() & nominator_bond_more() using the new arg flag unfreeze_bond: bool.

And as suggested introduced new api `unbond_frozen()`, to with draw the amount.